chain/bitcoind_client.go

```diff
@@ -1205,16 +1205,47 @@ func (c *BitcoindClient) filterTx(tx *wire.MsgTx,
 	// to determine if this transaction is somehow relevant to the caller.
 	var isRelevant bool
 
-	// We'll start by cycling through its outputs to determine if it pays to
+	// We'll start by checking all inputs and determining whether it spends
+	// an existing outpoint or a pkScript encoded as an address in our watch
+	// list.
+	for _, txIn := range tx.TxIn {
+		// If it matches an outpoint in our watch list, we can exit our
+		// loop early.
+		if _, ok := c.watchedOutPoints[txIn.PreviousOutPoint]; ok {
+			isRelevant = true
+			break
+		}
+
+		// Otherwise, we'll check whether it matches a pkScript in our
+		// watch list encoded as an address. To do so, we'll re-derive
+		// the pkScript of the output the input is attempting to spend.
+		pkScript, err := txscript.ComputePkScript(
+			txIn.SignatureScript, txIn.Witness,
+		)
+		if err != nil {
+			// Non-standard outputs can be safely skipped.
+			continue
+		}
+		addr, err := pkScript.Address(c.chainParams)
+		if err != nil {
+			// Non-standard outputs can be safely skipped.
+			continue
+		}
+		if _, ok := c.watchedAddresses[addr.String()]; ok {
+			isRelevant = true
+			break
+		}
+	}
+
+	// We'll also cycle through its outputs to determine if it pays to
```

Comment:
Similarly, can't we skip this loop if `isRelevant` is set above?

Reply:
Nope, as `NotifyReceived` for outpoints will also track them for spends, so we still want that logic to happen for backwards compatibility.